Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm

```diff
@@ -262,41 +292,113 @@ static bool isCopyDisplayedPixelBufferAvailable()
 
 MediaTime MediaPlayerPrivateWebM::currentMediaTime() const
 {
-    MediaTime synchronizerTime = PAL::toMediaTime(PAL::CMTimebaseGetTime([m_synchronizer timebase]));
+    MediaTime synchronizerTime = clampTimeToLastSeekTime(PAL::toMediaTime(PAL::CMTimebaseGetTime([m_synchronizer timebase])));
     if (synchronizerTime < MediaTime::zeroTime())
         return MediaTime::zeroTime();
-    if (synchronizerTime > durationMediaTime())
-        return durationMediaTime();
+    if (synchronizerTime < m_lastSeekTime)
```

Comment:
Isn't it redundant with what clampTimeToLastSeekTime does?

Reply:
Probably, but again, this is copying the logic from MediaPlayerPrivateMediaSourceAVFObjC and I don't want to modify MediaPlayerPrivateMediaSourceAVFObjC at this late stage.